### PR TITLE
Chore/create image placeholder

### PR DESCRIPTION
**Pull request by JamesRobertSutcliffe**
- Code added to render logo as placeholder if user does not have profile picture - can you please test this as I believe you do not have a profile picture saved? 

Also some general cleanup with re-optimisation for typescript.

### CODE REVIEW

File: src/app/personalised-homepage/page.js [hunk 1]
```diff
@@ -83,10 +83,15 @@ function PersonalisedHomepage() {
         return userProduct === "premium"
     }
 
+    // UserProfilePic function checks whether user 
+
+    const UserProfilePic = () => {
+        return userImage !== null
```
Comment (patdel0) [inline src/app/personalised-homepage/page.js:89]: When the user doesn't have an image, `userImage` becomes an empty string. 

This means that this statement will always return true, regardless of the user having a profile picture or not.

```js
"" !== null // returns true
```

**Review by patdel0 — APPROVED** [review]

**Review by patdel0 — CHANGES_REQUESTED** [review]
It's not working as expected. 
With the `!== null` condition, it doesn't display the image at all. 

When the condition is changed, the image appears for a couple of seconds, then disappears. This suggests to me that there is a state that is changing the value of `userImage`.

Happy to take over this one if you want. It would be a good opportunity to learn and create some tests for this one though.

**Comment by JamesRobertSutcliffe** [comment]
Thanks for this, I will make the change to this and try and write some tests!

**Comment by JamesRobertSutcliffe** [comment]
I have added some commits. Here is what I have done / tried to achieve - 

1- I have created a placeholder "Loading" component that appears until profile data is fetched. This stops the image from flipping. I intend to change the loading text to a slicker loading animation in the future
2 - I have changed the dependency on displaying the placeholder logo to on the data fetched being undefined - hopefully this fixes. 
3 - I have created tests in the 'profilePic.spec.ts' file - I was a little unsure what I was doing here or if I did it correctly so not sure how useful they are
4 - Please also test using your profile - hopefully the previous issues of logo displaying innappropriately / flicking between logo / profile pic is not fixed

I have also noticed that there is not a deployment issue. I do not have time to fix now. Let me know if you would prefer I fix this before checking, if not I can try and fix after you text / review.

File: tests/Frontend/profilePic.spec.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+test("Users profile picture should display if present", () => {
+    const userImage = "https://i.scdn.co/image/ab6775700000ee85357ee0eabf98dc740c83bb9c"
+    const setProfilePic = userImage === undefined ? "https://i.ibb.co/WHfbS7L/logo.png" : userImage;
+    expect(setProfilePic).toBe(userImage);
```
Comment (patdel0) [inline tests/Frontend/profilePic.spec.ts:4]: Nice one on starting out with Jest and testing! 👍

**No Action required** 

Right now, the test just checks if JavaScript and its ternary operators are doing their thing. It's not really looking at the component we're trying to test.

To focus on the component, you might want to test the actual DOM elements:

```js
import { render, screen } from '@testing-library/react';
import PersonalisedHomepage from '../PersonalisedHomepage';

it('should show the profile image', () => {
  render(<PersonalisedHomepage />);

  const imgElement = screen.getByRole('img');

  expect(imgElement).toBeInTheDocument();
  expect(imgElement).toHaveAttribute('src', 'expected-image-url');
});
```
To make this work, you'll need to add a package like `@testing-library/react` for DOM testing with Jest.

Here we are testing the front end and what the component renders in the document.

**Review by patdel0 — APPROVED** [review]
Nice job making it work! 

image

File: tests/Frontend/profilePic.spec.ts (hunk 2 above)
Comment (patdel0) [reply tests/Frontend/profilePic.spec.ts:4]: This code is for demonstration purposes only.

**Pull request merged**